src/Homework8/TTT.java

```diff
@@ -0,0 +1,164 @@
+            int j = (int) ((float)y/getHeight()*3);
+             // проверяем пуста ли выбранная ячейка,  и можно ли туда пойти
+            if (field[i][j] == DOT_EMPLY){
+             // проверка хода, если х то ставим крестик , если 0 то ставим нолик
+                field [i][j]= isXturn?DOT_X:DOT_0;
+                isXturn =!isXturn;// меняем ход
+                repaint();// перерисовка компоненка, вызывается paintComponent
+                int res = checState();
+                if (res!=0){
+                    //победили нолики
+                    if (res == DOT_0*3){
+                        JOptionPane.showMessageDialog(this,"выиграли нолики", "победа", JOptionPane.INFORMATION_MESSAGE);
+                    // победили крестики
+                    }else if (res == DOT_X*3){
+                        JOptionPane.showMessageDialog(this,"выиграли крестики", "победа", JOptionPane.INFORMATION_MESSAGE);
+                    }else {
+                        JOptionPane.showMessageDialog(this,"ничья","ничья", JOptionPane.INFORMATION_MESSAGE);
+                    }
+                    // перезапускаем игру
+                    initGame();
+                    //перезапускаем поле
+                    repaint();
+
+
+                }
+            }
+        }
+    }
+    @Override
+    protected void paintComponent(Graphics graphics){
+        super.paintComponent(graphics);
+        //очищаем холст
+        graphics.drawRect(0, 0, getWidth(),getHeight());
+        // рисуем сетку из линий
+       drawGrid(graphics);
+       //рисуем крестики и нолики
+       drawX0(graphics);
+    }
+
+
+
+    void drawGrid(Graphics graphics){
+        int w = getWidth();// ширина игрового поля
+        int h = getHeight();// высота игрового поля
+        int dw = w/3; // делим ширину поля на 3, получаем ширину 1й ячейки
+        int dh = h / 3;//делим высоту поля на 3, получаем высоту 1й ячейки
+        graphics.setColor(Color.BLACK); // цвет линий
+        for (int i=1;i<3;i++){// i пробегает значение от 1 до 2 включительно, при i=3 выход из цикла
+            graphics.drawLine(0,dh*i,w,dh*i);//горизонтальная линия
+            graphics.drawLine(dw*i,0,dw*i, h);// вертикальная линия
+        }
+
+
+
+    }
+    void drawX(int i, int j, Graphics graphics){
+        graphics.setColor(Color.BLACK);
+        int dw = getWidth()/3;
+        int dh = getHeight()/ 3;
+        var y = j * dh;
+        var x = i * dw;
+        graphics.drawLine(x,y,x+dw, y+dh);//линия от верхнего левого угла в правый нижний
+        graphics.drawLine(x, y+dh, x+dw,y);// линия от нижнего левого угла в правый верхний
+
+    }
+
+
+    void draw0(int i, int j, Graphics graphics) {
+         graphics.setColor(Color.BLACK);
+        int dw = getWidth()/ 3;
+        int dh = getHeight()/ 3;
+        var x = i * dw;
+        var y = j * dh;
+        graphics.drawOval(x * 5 * dw / 100, y, dw * 9 / 10, dh); // данное умножение необходимо для того, чтобы стенки
```

Comment:
слишком сложно, в итоге в последних ячейках овал не рисуется из-за данных умножений, таких вычислений будет достаточно
```suggestion
        graphics.drawOval(x + 5, y + 5, dw - 10, dh - 10);
```